### ISSUE TITLE
Search should throw warning if doc_type is specified for ES7+

### ISSUE THREAD

**Issue by EmilBode** [elastic/elasticsearch-py#1105]
SInce Elasticsearch 7.0, types are removed. In some cases, specifying a type as `_doc` is still allowed, but all other values are invalid.
Therefore, I think that if the user _does_ specify a type other than `_doc`, some warning should be given or an exception raised.
I encountered this when migrating from version 6 to 7, and finding out that when I removed the type in my data, my search queries did not fail, but instead just reported 0 results.

Reprex:
```
es.search(index='my-index', body='{"query": {"match_all": {}}}', doc_type='doc')
```
This worked fine in ES6 (assuming docs where stored with type `doc`). In ES7 this gives zero results, but no indication anything was wrong.
Note that
`es.index('my-index', body='{}', doc_type='doc')` _does_ fail, with a clear indication.

As there is no conveibale use-case to specify a doc_type (other than `_doc`), I think any value set to something else than `_doc` should at least print a waring, but preferably raise an exception.

**Environment**
Elasticsearch 7.5.1
elasticsearch-py 7.5.1 (from https://pypi.org/project/elasticsearch/)
Python 3.7.1 64-bit under Anaconda distribtion
ES running under Ubuntu 18.04.3 LTS
Python running under Windows 10

**Comment by AntonFriberg** [comment]
On a similar note the documentation for the helpers still contains the `"_type": "document"` which caused me some issues and should also be removed.

**Comment by sethmlarson** [comment]
This will be solved in 7.7 by raising warnings when Elasticsearch returns a `Warning` HTTP header.
